### DISCUSSION TITLE
Activate with right-click

### DISCUSSION THREAD

**Discussion by alessio89g** [extesy/hoverzoom#709] [Ideas]
Hi, great extension, very helpful and fast!
I think it has only one lack: activate the pop up with right-click mouse button.
Can you add this feature?

**Comment by extesy** [comment]
This suggestion seems to go against the purpose of the extension. You can't zoom on hover (see the extension name) and at the same time require right-clicking. If you want to right click, you can already pick "open image in new tab" option there.

**Reply by alessio89g** [reply]
Obviously it's your extension, mine was only a request.
However, I was previously using another extension for zoom on hover with a right-click, for years. Its name was [Imagus](https://chrome.google.com/webstore/detail/imagus/immpkjjlgappgfkkfieppnmlhakdmaab), but (I think) because of recents Chrome changes, it doesn't work correctly anymore.

**Reply by EhsanKia** [reply]
I'm also really interested in this feature, coming from Imagus (after seeing your HN comment). I know at first it sounds weird, but once you try it, it is really a well designed UX for viewing long videos and large images. Opening in a new tab/window is decent, but it's a context switch. Being able to zoom and pan around a photo inline is super useful. Hovering also is annoying for videos because if your mouse slips a bit, you lose your progress. The right click basically "locks" the media in the center, and allows you to move your mouse around. You then unlock it by clicking on the background or ESC.

If you're at least open to the idea, I'd be happy to prototype an implementation as a PR for you to play with and see how you feel about it.

**Reply by extesy** [reply]
@EhsanKia Thank you for the detailed explanation, now I better understand the original request. I am open to the idea, so please feel free to send a pull request, but please make sure that the new functionality is opt-in via options page and is turned off by default.

**Reply by Im-Baked** [reply]
I am really interested in this feature as well, will it ever get put in? I'm so used to using it from the old Hover Zoom right click on the image that is showing the big size and save image as. Imagus has it but that plugin doesn't work right in Chrome. Thanks for any information. Is there 1 I can download now somewhere?

**Reply by EhsanKia** [reply]
I already implemented this. Go to settings > action keys > "Lock image on screen" and set it to right click.

**Reply by Im-Baked** [reply]
Awesome thanks for the reply! I've been trying for days to get it to work
right. Even tried that version you put on the site but it was giving me an
error. I just couldn't figure this 1 out, so used to it just working by
itself from the old version of HoverZoom. 😀  I couldn't use that Imagus
plugin it didnt right with any site I was trying.

**Reply by EhsanKia** [reply]
What error are you getting, what version do you have?

The latest version should be 1.0.189, what are you doing and what doesn't work?

**Reply by Im-Baked** [reply]
No idea what the error was. It would just show it in the extensions tab. Deleted it. The version from the 1 I saw you put somewhere wasn't the latest version like this 1 is.  I have the latest version of this Hoverzoom+ 1.0.189. This version all work great now with what you told me to do in the options. But sometimes it doesn't wanna load the picture if I put the mouse over it, it will start to load it then just doesn't show it.

**Reply by HookCode** [reply]
> What error are you getting, what version do you have?
> 
> The latest version should be 1.0.189, what are you doing and what doesn't work?

Right click is not valid, if you use the keyboard shortcut is effective, the
Imagus right-click is effective, I turned off Imagus when using hoverzoom

**Reply by EhsanKia** [reply]
Right click works fine for me. I specifically coded it to work with right click because that's what I used on Imagus.

![image](https://user-images.githubusercontent.com/1619970/224599588-41ea8404-6e6d-4449-a791-e51bd3e7d739.png)

**Reply by LiliaDoe** [reply]
WIth the recent update, holding right click can be used to activate HZ+. This should solve your issue @alessio89g.
